File: Solutions/0506- Relative Ranks/Solution.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        temp = score[:]
        answer = [None] * len(score)
        rank_map = {}

        
        for index, value in enumerate(sorted(score, reverse=True)):
            rank_map[value] = index + 1

       
        for i in range(len(score)):
            rank = rank_map[temp[i]]
            if rank == 1:
                answer[i] = "Gold Medal"
            elif rank == 2:
                answer[i] = "Silver Medal"
            elif rank == 3:
                answer[i] = "Bronze Medal"
            else:
                answer[i] = str(rank)

        return answer
        

File: Solutions/0506- Relative Ranks/Solution.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
        int[] temp = score.clone();
        String[] answer = new String[score.length];
        Map<Integer, Integer> map = new HashMap<>();

        Arrays.sort(score);
        int rank = 1;

        for (int i = score.length - 1; i >= 0; i--) {
            map.put(score[i], rank);
            rank++;
        }

        for (int i = 0; i < score.length; i++) {
            int value = map.get(temp[i]);
            if (value == 1)
                answer[i] = "Gold Medal";
            else if (value == 2)
                answer[i] = "Silver Medal";
            else if (value == 3)
                answer[i] = "Bronze Medal";
            else
                answer[i] = String.valueOf(value);
        }

        return answer;

    }
}